valintalaskenta-laskenta-service/src/main/java/fi/vm/sade/valintalaskenta/laskenta/resource/LaskentaSeurantaResource.java

```diff
@@ -0,0 +1,449 @@
+package fi.vm.sade.valintalaskenta.laskenta.resource;
+
+import static fi.vm.sade.valintalaskenta.tulos.roles.ValintojenToteuttaminenRole.OPH_CRUD;
+
+import fi.vm.sade.javautils.nio.cas.CasClient;
+import fi.vm.sade.valintalaskenta.laskenta.resource.dto.*;
+import fi.vm.sade.valintalaskenta.tulos.RestClientUtil;
+import java.net.URLEncoder;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import org.asynchttpclient.Request;
+import org.asynchttpclient.RequestBuilder;
+import org.asynchttpclient.Response;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@PreAuthorize("isAuthenticated()")
+@RequestMapping(value = "/resources/seuranta")
+public class LaskentaSeurantaResource {
+
+  private CasClient seurantaCasClient;
+
+  private String seurantaBaseUrl;
+
+  public LaskentaSeurantaResource(
+      @Qualifier("seurantaCasClient") CasClient seurantaCasClient,
+      @Value("${valintalaskentakoostepalvelu.seuranta.rest.url}") String seurantaBaseUrl) {
+    this.seurantaCasClient = seurantaCasClient;
+    this.seurantaBaseUrl = seurantaBaseUrl;
+  }
+
+  public static org.asynchttpclient.Response execute(
+      final CasClient casClient, final Request request) {
+    try {
+      return casClient.executeBlocking(request);
+    } catch (final ExecutionException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /** Yhteenvedot olemassa olevista laskennoista haulle */
+  @PreAuthorize(OPH_CRUD)
+  @GetMapping(value = "/hae/{hakuOid}", produces = MediaType.APPLICATION_JSON_VALUE)
+  ResponseEntity<String> hae(@PathVariable("hakuOid") String hakuOid) {
```

Comment:
Onko tähän myöhemmässä tehtävässä tarkoitus laittaa jokin dto?

Reply:
Kyllä, juuri näin. Tässä välitilassa muunnos on turha.